ui/perfherder/graphs/GraphsContainer.jsx

```diff
@@ -417,6 +425,16 @@ class GraphsContainer extends React.Component {
                       />
                     ))}
 
+                  {changelogData.length > 0 && (
+                    <VictoryAxis
+                      tickValues={changelogData.map(i => i.date)}
+                      style={{
+                        tickLabels: { display: 'none' },
+                        grid: { stroke: '#ff0000' },
+                      }}
+                    />
+                  )}
```

Comment:
@sarah-clements Do you have any thoughts on this part? `<VictoryAxis>` seems like the best way to draw the red lines to indicate CI changes (`<VictoryLine>` and `<VictoryBar>` mess with the Y axis range of the graph, which I'd rather avoid). However, I'll have to do something clever to get this to show tooltips since `<VictoryAxis>` isn't designed to hold non-numeric data. Maybe I should just make my own Victory component that works roughly like `<VictoryAxis>`?

One other option might be to explicitly calculate the Y axis range for the graph and then use that to draw lines of the right size via `<VictoryBar>`. (`<VictoryLine>` doesn't seem to work too well, since it's built to draw a single continuous line; I think I'd have to use 1 `<VictoryLine>` per CI change.)

I'm not super-familiar with Victory so maybe I'm just going about this the wrong way...

(Note: if you want to try this out locally, just go to `localhost:5000/perf.html` and hover over one of the jobs and then click the "graph" link. It should show a bunch of red lines superimposed on the main graph.)

Reply:
I think using `<VictoryLine>` is the right approach. I tried changing your code - implementing it the way highlights is on line 416 - and I'm not seeing any Y axis issues. Can you elaborate on what problems you're seeing? Keep in mind, there is zooming on that graph - so if you accidentally select an area on the scatterplot, it will zoom it and add the `zoom` params to the url (so that will indeed change the y axis depending on the selection area for the zoom).

Please also use `yarn lint --fix` frequently during development. There's currently a compile issue `Module not found: Error: Can't resolve './GraphChangelogTooltip' in '/Users/sclements/Desktop/Mozilla/treeherder/ui/perfherder/graphs'` that I had to resolve before I could test this locally. Steps to follow are here:https://treeherder.readthedocs.io/testing.html#validating-javascript

Reply:
Oh, whoops. I fixed the lint issue locally but must have forgotten to push it. That should be fixed now.

Reply:
@sarah-clements Here are some screenshots that hopefully show the issue I was having with the Y axis range:

First, here's my patch using `<VictoryAxis>`:
![axis-nohighlight](https://user-images.githubusercontent.com/826865/78597072-b37f8280-7801-11ea-94ce-3325f19e2a3d.png)

Now, here's a patch using `<VictoryLine>` (I just create one line component per datapoint in a loop, using the same definition as the highlighted test run on line 416):
![line-nohighlight](https://user-images.githubusercontent.com/826865/78597159-d9a52280-7801-11ea-9803-3b01cdd2105c.png)

One complication here is that if you have a highlighted test run, the code rendering that (line 416) *also* messes with the Y axis:
![highlight-only](https://user-images.githubusercontent.com/826865/78597307-21c44500-7802-11ea-8871-6e43acd9844c.png)

I much prefer the first screenshot since it's considerably easier to see the difference between various test results. If you think it would make sense, I could also use the `<VictoryAxis>` method on the highlighted test run so that it doesn't mess with the Y axis range (I would file a separate PR for that though, since it's not directly related to this one).

Reply:
hrm, this is what I see:

with VictoryAxis
![victory_axis](https://user-images.githubusercontent.com/19615783/78607412-de72d200-7813-11ea-8baf-f550fe00da16.png)

using VictoryLine
![victory_line](https://user-images.githubusercontent.com/19615783/78607430-e3378600-7813-11ea-8811-fc40c4c91934.png)


```                  {changelogData.length > 0 && 
                    changelogData.map(item => (
                    <VictoryLine
                      x={() => item.date}
                      style={{
                        data: { stroke: '#ff0000', strokeWidth: 1 },
                      }}
                    />
                  ))}
```
Can you share the links for the graph that looks normal and the one that looks wonky? I'm wondering if something else is going on.

Reply:
Normal: http://localhost:5000/perf.html#/graphs?highlightAlerts=0&series=autoland,2247031,1,1&timerange=1209600
Wonky: http://localhost:5000/perf.html#/graphs?highlightAlerts=1&series=autoland,2247031,1,1&timerange=1209600

The "Normal" link should look ok on master and with the `<VictoryAxis>` implementation, but not the `<VictoryLine>` implementation. You can also switch between these variants by clicking the "Highlight alerts" button below the graph.

Reply:
I think I've figured out a work-around to the y-axis issue with the `VictoryLine` component. You need to set `domain={y: [minValue, maxValue]}` which you could get from the `testData` (although keep in mind it could be an array of multiple objects) or the `tickValues` from the first `VictoryAxis` but that's a bit tricky to store in state (maybe debounced would work?). We use lodash so feel free to pull in any helpers.

This problem also only seems to work when not zoomed, so it probably only needs to update when test data is changed (store it in state) if you go this route.

However, I'm not opposed to you using the `VictoryAxis` (and switching highlight alerts in this pr), but that might be more work to figure out how to access the data you need and to then display a tooltip.

Also, you'll need to convert the date to UTC: `moment.utc(date).format('MMM DD')` or use `this.checkDate`

One minor suggestion - can you make the red a softer opacity. Seems a bit harsh :)

Reply:
Another thought I've had is to not show a line but to find any data points that coincide with a changelog date and add some info to that tooltip "infrastructure changes related to this - click more for details" or something like that. Either way,  might be a useful to get feedback from @ionutgoldan about how to implement this feature in a way that will be most useful (if you haven't already). :)

Reply:
Yeah, explicitly setting the `domain` for the Y values would be one solution, tough I hoped there was a more-automatic way to keep things simple...

I pinged @ionutgoldan to get some feedback though, since there might be a nicer way to present this info. And I agree on changing the color; I usually just use red when I'm trying things out since it stands out from the rest of the UI. :)